src/cargo/core/compiler/custom_build.rs

```diff
@@ -716,10 +725,9 @@ impl BuildOutput {
                         if nightly_features_allowed
                             || rustc_bootstrap_allows(library_name.as_deref())
                         {
-                            warnings.push(format!("Cannot set `RUSTC_BOOTSTRAP={}` from {}.\n\
+                            diagnostics.push(Diagnostic::Warning(format!("Cannot set `RUSTC_BOOTSTRAP={val}` from {whence}.\n\
                                 note: Crates cannot set `RUSTC_BOOTSTRAP` themselves, as doing so would subvert the stability guarantees of Rust for your project.",
-                                val, whence
-                            ));
+                            )));
```

Comment:
In the future, please do not mix unrelated refactors in the same commit

Reply:
I've changed these to match other lines that I had to change, because rustfmt made a mess of them (the indentation level here is pretty high).